addons/stock/models/stock_production_lot.py

```diff
@@ -105,3 +106,37 @@ def action_lot_open_quants(self):
         if self.user_has_groups('stock.group_stock_manager'):
             self = self.with_context(inventory_mode=True)
         return self.env['stock.quant']._get_quants_action()
+
+    def _create_or_update_sn_conflict_inv(self):
+        """ Adds/removes lot_ids from an existing (draft) duplicate SN inventory by company.
+        If in progress dupe sn inventory exists, then nothing will be done. If no draft/in progress
+        inventory exists, then a new one will be created. If no lots remain in the (draft) inventory
+        then it is auto-deleted.
+        """
+        Inventory = self.env['stock.inventory']
+        company_id = self.company_id
+        domain = [('lot_ids', '!=', False), ('company_id', '=', company_id.id)]
+        draft_inv = Inventory.search(expression.AND([[('state', '=', 'draft')], domain]), limit=1)
+        quants = self.env['stock.quant'].search([('location_id.usage', 'in', ['internal', 'transit']),
+                                                 ('lot_id', '=', self.id),
+                                                 ('product_id.tracking', '=', 'serial'),
+                                                 ('quantity', '!=', 0.0),
+                                                 ('company_id', '=', company_id.id)])
+        if len(quants) > 1:
```

Comment:
As you don't need quants data, you can use search_count instead of search. It will be quicker